src/Language/PureScript/Parser/Types.hs

```diff
@@ -81,9 +81,11 @@ parseConstrainedType = do
     indented
     ty <- P.many parseTypeAtom
     return (Constraint className ty Nothing)
+  oldComma = comma *>
+    P.parserFail "Class constraints in type annotations can no longer be grouped in parentheses.\n  Each constraint should now be separated by `=>`, for example:\n    `(Applicative f, Semigroup a) => a -> f a -> f a`\n  would now be written as:\n    `Applicative f => Seimgroup a => a -> f a -> f a`."
 
 parseAnyType :: TokenParser Type
-parseAnyType = P.buildExpressionParser operators (buildPostfixParser postfixTable (P.try parseConstrainedType <|> parseTypeAtom)) P.<?> "type"
+parseAnyType = P.buildExpressionParser operators (buildPostfixParser postfixTable (P.try parseConstrainedType <|> P.try parseTypeAtom <|> parseConstrainedType)) P.<?> "type"
```

Comment:
This is gross, but aside from using `error` or something, I'm not sure how else to get the new message to appear.